packages/editor/src/components/browser/TypeBrowser.tsx

```diff
@@ -160,16 +184,32 @@ const TypeBrowser = ({ value, onChange, onDoubleClick, initSearchFilter }: TypeB
     };
 
     if (Object.keys(rowSelection).length !== 1) {
-      onChange('');
+      onChange({ cursorValue: '' });
       setShowHelper(false);
       return;
     }
-    let selectedRow = undefined;
-    selectedRow = tableDynamic.getRowModel().rowsById[Object.keys(rowSelection)[0]];
-
+    const selectedRow = tableDynamic.getRowModel().rowsById[Object.keys(rowSelection)[0]];
+    setHelperText(addListGeneric(selectedRow.original, typeAsList));
     setShowHelper(true);
-    onChange(addListGeneric(selectedRow.original, typeAsList));
-  }, [onChange, rowSelection, tableDynamic, typeAsList]);
+    if (location.includes('code')) {
+      let cursorValueShort = helperText.split('.').length > 1 ? String(helperText.split('.').pop()) : helperText;
+      if (typeAsList) {
+        const lastIndexBeforeBracket = helperText.lastIndexOf('<');
+        const lastIndexBeforePeriod = helperText.lastIndexOf('.', lastIndexBeforeBracket);
+        cursorValueShort = lastIndexBeforePeriod !== -1 ? helperText.substring(lastIndexBeforePeriod + 1) : helperText;
+      }
+      if (!ivyTypes.some(javaClass => javaClass.fullQualifiedName === selectedRow.original.fullQualifiedName) && !typeAsList) {
+        onChange({
+          cursorValue: cursorValueShort,
+          firstLineValue: 'import ' + helperText + ';\n'
+        });
+      } else {
+        onChange({ cursorValue: cursorValueShort });
+      }
+    } else {
+      onChange({ cursorValue: helperText });
+    }

```

Comment:
@ivy-lli What do you think about this part of the code? I find it a bit complicated, but I don't know, how to solve it differently. With this code, the import works much better than before.

Reply:
Don't understand the code here...
why do you need to extract the values from the helpertext and no longer from the full row with all the different properties?
And in general, I think this code here should be in a separate function and should have tests... Would also be much easier to read this logic and to check that it always does what we are expecting.